core/src/main/java/com/redhat/lightblue/client/response/BulkLightblueResponse.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * 
+ */
+package com.redhat.lightblue.client.response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.model.DataError;
+import com.redhat.lightblue.client.model.Error;
+import com.redhat.lightblue.client.request.AbstractBulkLightblueRequest;
+import com.redhat.lightblue.client.request.AbstractLightblueRequest;
+import com.redhat.lightblue.client.request.LightblueRequest;
+import com.redhat.lightblue.client.util.JSON;
+
+/**
+ * @author bvulaj
+ *
+ */
+public class BulkLightblueResponse {
+
+    private final List<LightblueResponse> responses;
+    private final List<? super AbstractLightblueRequest> requests;
+
+    private JsonNode json;
+    private String text;
+
+    public BulkLightblueResponse(List<LightblueResponse> resps, AbstractBulkLightblueRequest<? extends AbstractLightblueRequest> reqs) {
+        responses = new ArrayList<LightblueResponse>(resps);
```

Comment:
That's the response, why add()?


On Wed, Sep 9, 2015 at 9:27 AM, Brandon Vulaj <notifications@github.com>
wrote:

> In
> core/src/main/java/com/redhat/lightblue/client/response/BulkLightblueResponse.java
> <https://github.com/lightblue-platform/lightblue-client/pull/156#discussion_r39057266>
> :
>
> > +import com.redhat.lightblue.client.util.JSON;
> > +
> > +/**
> > + * @author bvulaj
> > + *
> > + */
> > +public class BulkLightblueResponse {
> > +
> > +    private final List<LightblueResponse> responses;
> > +    private final List<? super AbstractLightblueRequest> requests;
> > +
> > +    private JsonNode json;
> > +    private String text;
> > +
> > +    public BulkLightblueResponse(List<LightblueResponse> resps, AbstractBulkLightblueRequest<? extends AbstractLightblueRequest> reqs) {
> > +        responses = new ArrayList<LightblueResponse>(resps);
>
> @bserdar <https://github.com/bserdar> This was to support add() since not
> all List impls support it. ie -
> http://docs.oracle.com/javase/6/docs/api/java/util/Arrays.html#asList%28T...%29
>
> I guess we don't really need to support add - I'll refactor slightly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/156/files#r39057266>
> .
>